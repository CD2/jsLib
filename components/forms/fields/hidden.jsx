import React from "react"
import PropTypes from "prop-types"

export class TextField extends React.Component {
  static propTypes = {
    initialValue: PropTypes.string,
    name: PropTypes.string,
    onChange: PropTypes.func,
    onFocus: PropTypes.func,
    onKeyPress: PropTypes.func,
    onRawChange: PropTypes.func,
    placeholder: PropTypes.string,
    type: PropTypes.string,
  }

  handleChange = e => {
    const { onRawChange, onChange } = this.props
    if (onRawChange) onRawChange(e)
    if (onChange) onChange({ name: e.target.name, value: e.target.value })
  }

  render() {
    const props = { ...this.props }
    delete props.initialValue
    delete props.onRawChange
    delete props.theme
    props.type = props.type || `text`
    props.value = props.value || ``

    return <input {...props} onChange={this.handleChange} />
  }
}
export default TextField
