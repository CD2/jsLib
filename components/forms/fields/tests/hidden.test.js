/* eslint-disable react/jsx-filename-extension */
import React from 'react'
import { TextField } from '../hidden'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

const wrapper = shallow(<TextField />)
it(`snapshot`, ()=>expect(toJson(wrapper)).toMatchSnapshot())
