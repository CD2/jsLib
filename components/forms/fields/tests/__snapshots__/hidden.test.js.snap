// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<input
  onChange={[Function]}
  type="text"
  value=""
/>
`;
